### PR TITLE
Sets expectations for testcaserun predicates

### PR DESCRIPTION
**Pull request by dan-han-101**
# Changes

The documentation update sets expecations on the sequence of expected test events.

This update was triggered by discussions in
https://github.com/cdevents/spec/issues/140

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has the [primer doc](https://github.com/cdevents/spec/blob/main/primer/_index.md) been updated if a design decision is involved
- [x] Have the [JSON schemas](https://github.com/tektoncd/community/blob/main/standards.md#tests) been updated if the specification changed
- [x] Has spec version and event versions been updated according to the [versioning policy](https://github.com/cdevents/spec/blob/main/primer/_index.md#versioning)
- [x] Meets the [CDEvents contributor standards](https://github.com/cdevents/.github/blob/main/docs/CONTRIBUTING.md)

### CODE REVIEW

**Review by olensmar — APPROVED** [review]

File: testing-events.md [hunk 1]
```diff
@@ -16,7 +16,11 @@ Testing events covers the subjects and predicates related to test-execution perf
 
 ## Subjects
 
-This specification defines three subjects in this stage: `testCaseRun`, `testSuiteRun` and `testOutput`.
+This specification defines three subjects in this stage: `testCaseRun`, `testSuiteRun` and `testOutput`. When a producer sends `testCaseRun` or `testSuiteRun` messages, they should meet the following expecations:
+- a `queued` event may optionally be followed by a `started` event
+- a `queued` event must be followed by a `finished` event
+- a `started` event must be followed by a `finished` event
+- a `finished` event may be sent independent of a `queued` or `started` event
```
Comment (e-backmark-ericsson) [inline testing-events.md:23]: I would rather see something like this:
```suggestion
- a `queued` event may optionally be followed by a `started` event
- a `queued` event must be followed by either a `started` or a `finished` event
- a `finished` event following after a `queued` event without a `started` event in between, can only have its outcome set to `cancel` (or `skip` if that is available in the protocol)
- a `started` event must be followed by a `finished` event
- a `finished` event may not be sent if no `queued` or `started` events are sent for the same subject
```

**Review by afrittoli — COMMENTED** [review]
Thanks @dan-han-101 - this makes sense to me. 
I guess we should merge the PR that adds "skip" first?

File: testing-events.md [hunk 2]
```diff
@@ -16,7 +16,12 @@ Testing events covers the subjects and predicates related to test-execution perf
 
 ## Subjects
 
-This specification defines three subjects in this stage: `testCaseRun`, `testSuiteRun` and `testOutput`.
+This specification defines three subjects in this stage: `testCaseRun`, `testSuiteRun` and `testOutput`. When a producer sends `testCaseRun` or `testSuiteRun` messages, they should meet the following expecations:
```
[the remark this answers is not in the document]
Comment (dan-han-101) [reply testing-events.md:19]:
```suggestion
This specification defines three subjects in this stage: `testCaseRun`, `testSuiteRun` and `testOutput`. When a producer sends `testCaseRun` or `testSuiteRun` messages, they should meet the following expectations:
```

**Comment by dan-han-101** [comment]
Sure, we can wait for the other PR to complete first

**Comment by dan-han-101** [comment]
FYI - my previous commits were failing the DCO checks. 

In order to fix them, I had to do a force push where I updated the author email to be more consistent, and then added the sign off message in each commit.

**Review by afrittoli — APPROVED** [review]
Thank you! LGTM

**Comment by dan-han-101** [comment]
Closing this one, in favor of https://github.com/cdevents/spec/pull/145/files . 
That update has the schema/spec changes + the documentation changes in one PR.

**Pull request closed without merging**